src/metaopt/core/worker/trial.py

```diff
@@ -134,7 +134,7 @@ class Result(Value):
         function or of an 'constraint' expression.
         """
 
-        allowed_types = ('objective', 'constraint')
+        allowed_types = ('objective', 'constraint', 'gradient')
```

Comment:
For example, a distributed asynchronous gradient descent algorithm would use the gradients with respect to the hyperparameters to optimize them.

Reply:
I understand that some meta-algo can use gradients, I was questioning the idea of specifically describing it as gradients rather than some generic type. But it's probably better to use gradient otherwise a generic type might be misused because of laziness, ahem, inertia. :wink: